ibis/backends/dask/tests/execution/test_timecontext.py

```diff
@@ -218,3 +231,88 @@ def test_context_adjustment_window_groupby_id(time_table, time_df3):
     # result should adjust time context accordingly
     result = expr.execute(timecontext=context)
     tm.assert_series_equal(result, expected)
+
+
+def test_adjust_context_scope(time_keyed_left, time_keyed_right):
```

Comment:
To avoid confusion, this new Dask backend test is not very related to this PR. This should have been mirrored from the Pandas tests in #3037, and I am just fixing this now, since this PR already needs to bring over the `CustomAsOfJoin` text fixtures from the Pandas tests anyways.

Reply:
I am still confused. Why are most of the tests in this PR under dask tests? How are we testing pandas backend?

Reply:
Oh nvm saw ot